### PR TITLE
Support for Loongson jdk build

### PR DESCRIPTION
**Pull request by Panxuefeng-loongson**
Hi,

I'm a member of the openjdk development team from Loongson. We intend to distribute binaries of  JDK via adoptium. LoongArch code was based on openjdk 8, here is repo on github:https://github.com/loongson/jdk8u. LoongArch is a risc architecture, developed by Loongson Technology. Currently, the freetype version needs to be updated, see: https://github.com/adoptium/temurin-build/issues/3120. So this pr can't be merged yet, it's just used to ask for your opinion in advance.

We intend to use jdk8 as a trial and look forward to participating in more adoptium community work.

Thanks

### CODE REVIEW

**Comment by karianna** [comment]
Hi @Panxuefeng-loongson - Are you able to sign the Eclipse CLA?

**Comment by karianna** [comment]
@Panxuefeng-loongson the Loongsoon build itself seems to be failing: https://github.com/adoptium/temurin-build/actions/runs/3327315068/jobs/5515528475

**Comment by Panxuefeng-loongson** [comment]
> @Panxuefeng-loongson the Loongsoon build itself seems to be failing: https://github.com/adoptium/temurin-build/actions/runs/3327315068/jobs/5515528475

I only tested it on my local machine,  I will fix the build failure in docker.

**Comment by Panxuefeng-loongson** [comment]
> Hi @Panxuefeng-loongson - Are you able to sign the Eclipse CLA?

I clicked on this link: https://accounts.eclipse.org/user/register?destination=legal/eca/validation/125041 and want to register an account, an error occurred while submitting information: The answer you entered for the CAPTCHA was not correct. Could you please tell me where to sign the cla agreement?

**Comment by smlambert** [comment]
This has some guidance for ECA signing, hopefully it can help: https://adoptium.net/docs/eca-sign-off

**Comment by Panxuefeng-loongson** [comment]
> Hi @Panxuefeng-loongson - Are you able to sign the Eclipse CLA?

I have signed the Eclipse CLA. Thanks for your help @karianna @smlambert

**Comment by smlambert** [comment]
ECA validation now passes 👍

File: README.md
Comment (karianna) [inline README.md:29]:
```suggestion
./makejdk-any-platform.sh (-J /usr/lib/jvm/jdk-xx) (--build-variant <hotspot|openj9|corretto|SapMachine|dragonwell|bisheng|loongson>) <jdk8u|jdk11u|jdk17u|jdk>
```

File: README.md
Comment (karianna) [inline README.md:371]:
```suggestion
This tag identifies the JVM being used by the JDK. "dragonwell", "bisheng", and "loongson" itself are HotSpot based JVMs but are currently considered their own variants for the purposes of build.
```

File: build-farm/make-adopt-build-farm.sh
Comment (karianna) [inline build-farm/make-adopt-build-farm.sh:34]: Can you also increase the whitespace indent for the `fi` parts of lines above and below so they line up again?

File: build-farm/platform-specific-configurations/linux.sh
Comment (karianna) [inline build-farm/platform-specific-configurations/linux.sh:145]: Can we have an https download and also a checksum check afterwawrds?

File: configureBuild.sh
Comment (karianna) [inline configureBuild.sh:177]: This seems to be coming from your repo, should it be from your company's repo?

File: sbin/build.sh
Comment (karianna) [inline sbin/build.sh:254]:
```suggestion
        echo "Only Java 8 is supported for now"
```

File: sbin/build.sh
Comment (karianna) [inline sbin/build.sh:323]: Would prefer to have some valid URLs here else it's hard for users to know where to report bugs

**Review by karianna — CHANGES_REQUESTED** [review]
Some minor changes, broadly looks good.

File: configureBuild.sh
Comment (Panxuefeng-loongson) [reply configureBuild.sh:177]: You're right. But the company code repo does not have a tag to meet the build requirements, using my own fork repo is only a temporary solution.

**Comment by Panxuefeng-loongson** [comment]
@theaoqi is our team leader, you can also ask him if you have any questions. Before merge this change, I would like to ask @theaoqi to confirm the changes again.

File: build-farm/platform-specific-configurations/linux.sh
Comment (karianna) [reply build-farm/platform-specific-configurations/linux.sh:145]: Is https available?

File: configureBuild.sh
Comment (karianna) [reply configureBuild.sh:177]: OK, we'll likely have to hold off until we have that in place to make sure there's no MitM attacks on the binary blob before it gets installed on one of our agents.

File: build-farm/platform-specific-configurations/linux.sh
Comment (theaoqi) [inline build-farm/platform-specific-configurations/linux.sh:138]: I suggest not using `elif`, just a new line and `if`

File: build-farm/platform-specific-configurations/linux.sh
Comment (theaoqi) [inline build-farm/platform-specific-configurations/linux.sh:139]: How about `$ARCHITECTURE` for code consistent.

File: build-farm/platform-specific-configurations/linux.sh
Comment (theaoqi) [inline build-farm/platform-specific-configurations/linux.sh:140]: 1. Do not consider the situation `-d /opt/j2sdk-images`
2. Consider the situation in which `$JAVA_FEATURE_VERSION` is not 8. We may add support for more java verison, like 17 and 11.

Comment (Panxuefeng-loongson) [reply build-farm/platform-specific-configurations/linux.sh:140]: The proposal seems reasonable. I will fix this issue.

File: sbin/build.sh
Comment (theaoqi) [inline sbin/build.sh:1402]: TAG_SEARCH="jdk8u*ls-*"

File: sbin/build.sh
Comment (theaoqi) [inline sbin/build.sh:322]: Loongson

File: configureBuild.sh
Comment (theaoqi) [reply configureBuild.sh:177]: I have pushed a new tag into loongson/jdk8u. It is ready for use now.

File: build-farm/platform-specific-configurations/linux.sh
Comment (karianna) [reply build-farm/platform-specific-configurations/linux.sh:145]: OK, we'll likely have to hold off until we have that in place to make sure there's no MitM attacks on the binary blob before it gets installed on one of our agents.

**Comment by Panxuefeng-loongson** [comment]
> Assume after we enable this feature (do nightly+weekly jdk8 build to support loongson), we need update ci-jenkins-pipeline as well

Thank you for the reminder. I will update ci-jenkins-pipeline soon.

**Comment by tellison** [comment]
This enhancement is being discussed by the PMC on Nov 9, 2022.
See also https://github.com/adoptium/infrastructure/issues/2806

File: build-farm/platform-specific-configurations/linux.sh
Comment (Panxuefeng-loongson) [reply build-farm/platform-specific-configurations/linux.sh]: > OK, we'll likely have to hold off until we have that in place to make sure there's no MitM attacks on the binary blob before it gets installed on one of our agents.

I hava changed http to https.

**Review by karianna — APPROVED** [review]
LGTM - Please note there is a separate decision whether to add build agents at Adoptium, but at least this should enable folks to build locally.

**Review by theaoqi — APPROVED** [review]
LGTM

**Review by theaoqi — COMMENTED** [review]
Looks like the patch broke the builds? @Panxuefeng-loongson please have a look.

**Comment by Panxuefeng-loongson** [comment]
Some errors have occurred, i will check it

File: build-farm/platform-specific-configurations/linux.sh
Comment (tellison) [inline build-farm/platform-specific-configurations/linux.sh:146]: Why is the architecture check required? Doesn't it require a Loongson JVM to build the Loonson codebase every time?

File: build-farm/platform-specific-configurations/linux.sh
Comment (tellison) [inline build-farm/platform-specific-configurations/linux.sh:156]: Roll into the variant check above?

File: build-farm/platform-specific-configurations/linux.sh
Comment (Panxuefeng-loongson) [reply build-farm/platform-specific-configurations/linux.sh]: > Why is the architecture check required? Doesn't it require a Loongson JVM to build the Loonson codebase every time?

I found that in jenkins jobs, Loongson codebase will build on x86 machine. If not architecture check, x86 machine will use the LoongArch jdk binary, and an error will occur when VARIANT is BUILD_VARIANT_LOONGSON. Maybe my ideas are wrong and welcome to give your opinion.

**Comment by Panxuefeng-loongson** [comment]
Please review the change again and welcome to give your opinion @tellison @karianna

**Review by andrew-m-leonard — APPROVED** [review]
Looks good, thanks

**Comment by karianna** [comment]
Couple of shellcheck warnings:

```
...
2022-11-26 06:18:35 [INFO]   File:[/github/workspace/sbin/build.sh]
2022-11-26 06:18:37 [ERROR]   Found errors in [shellcheck] linter!
2022-11-26 06:18:37 [ERROR]   Error code: 1. Command output:
------

In /github/workspace/sbin/build.sh line 250:
        local updateNum="$(cat "$loongsonVerFile" | awk -F - '{print $1}' | awk -F u '{print $2}')"
                               ^----------------^ SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In /github/workspace/sbin/build.sh line 251:
        local buildNum="$(cat "$loongsonVerFile" | awk -F - '{print $2}')"
                              ^----------------^ SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

For more information:
  https://www.shellcheck.net/wiki/SC2002 -- Useless cat. Consider 'cmd < file...
------
...
```

**Comment by Panxuefeng-loongson** [comment]
> Couple of shellcheck warnings:
> 
> ```
> ...
> 2022-11-26 06:18:35 [INFO]   File:[/github/workspace/sbin/build.sh]
> 2022-11-26 06:18:37 [ERROR]   Found errors in [shellcheck] linter!
> 2022-11-26 06:18:37 [ERROR]   Error code: 1. Command output:
> ------
> 
> In /github/workspace/sbin/build.sh line 250:
>         local updateNum="$(cat "$loongsonVerFile" | awk -F - '{print $1}' | awk -F u '{print $2}')"
>                                ^----------------^ SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
> 
> 
> In /github/workspace/sbin/build.sh line 251:
>         local buildNum="$(cat "$loongsonVerFile" | awk -F - '{print $2}')"
>                               ^----------------^ SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC2002 -- Useless cat. Consider 'cmd < file...
> ------
> ...
> ```

I will fix this problem, thanks

**Review by karianna — APPROVED** [review]
LGTM

**Comment by Panxuefeng-loongson** [comment]
ping

**Comment by gdams** [comment]
@Panxuefeng-loongson this looks good to me, on the topic of FreeType, I propose that we merge https://github.com/adoptium/temurin-build/pull/3188 first which would then allow us to specify a different FreeType tag for Loongson JDK. What do you think?

**Comment by Panxuefeng-loongson** [comment]
> @Panxuefeng-loongson this looks good to me, on the topic of FreeType, I propose that we merge #3188 first which would then allow us to specify a different FreeType tag for Loongson JDK. What do you think?

This option looks great and I agree with your solution

**Comment by Panxuefeng-loongson** [comment]
> Out of interest, what FreeType version does Loongson JDK need @Panxuefeng-loongson?

Recent versions of freetype have met the Loongson JDK build, because config.guess and config.sub support LoongArch from this point onwards: http://git.savannah.gnu.org/gitweb/?p=config.git;a=commit;h=c8ddc8472f8efcadafc1ef53ca1d863415fddd5f, It's been two years since. I have checked the last few versions of freetype and they all meet our needs. Thanks @gdams

**Comment by andrew-m-leonard** [comment]
this has been stale for over a year, converting to "draft", if no longer needed please close.

**Comment by sxa** [comment]
Closing on the basis that Temurin will not be integrating builds on this (And the PR would now require a bit of a rebase). If there is a need for this for people building outside temurin then it can be revisited.

**Pull request closed without merging**
